LayoutTests/fast/canvas/webgl/colorspaces-test.html

```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL Canvas Conformance Tests</title>
+<script src="../../../resources/js-test.js"></script>
+<script src="resources/webgl-test.js"></script>
+<script src="resources/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+
+<script>
+description("This test ensures WebGL implementations correctly implement drawingbufferColorSpace.");
+
+function near(a, b, epsilon) {
+  return (a >= b - epsilon) && (a <= b + epsilon);
+}
+
+function checkPixels(gl, w, h, color, alpha) {
+  pixelCount = w * h;
+  const pixels = new Uint8Array(pixelCount * 4);
+  gl.readPixels(0, 0, w, h, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+  correctPixels = 0;
+  for (i = 0; i < pixelCount; ++i) {
+    if (near(pixels[i * 4 + 0], color[0] * 255, 1)
+     && near(pixels[i * 4 + 1], color[1] * 255, 1)
+     && near(pixels[i * 4 + 2], color[2] * 255, 1)
+     && near(pixels[i * 4 + 3], alpha * 255, 1)) {
+      correctPixels += 1;
+    }
+  }
+  shouldBe('correctPixels', 'pixelCount');
+}
+
+const DONT_SET = "<don't set>";
+
+function test(w, h, drawingBufferColorSpaceInString, expectedDrawingBufferColorSpace, color) {
+  debug("");
+  debug(`test(w=${w}, h=${h}, drawingBufferColorSpace=${drawingBufferColorSpaceInString}, expectedDrawingBufferColorSpace="${expectedDrawingBufferColorSpace}", color="${color}")`);
+  const drawingBufferColorSpace = eval(drawingBufferColorSpaceInString);
+
+  var canvas = document.createElement("canvas");
+  canvas.width = w;
+  canvas.height = h;
+
+  var consoleDiv = document.getElementById("console");
+  document.body.insertBefore(canvas, consoleDiv);
+
+  // This 'gl' must be global for shouldBe to work.
+  gl = wtu.create3DContext(canvas, {antialias: false});
+  if (!gl) {
+    testFailed("context does not exist");
+  } else {
+    testPassed("context exists");
+
+    if ('drawingBufferColorSpace' in gl) {
+      shouldBe('gl.drawingBufferColorSpace', '"srgb"');
+
+      if (drawingBufferColorSpace !== DONT_SET) {
+        debug(`Setting gl.drawingBufferColorSpace=${drawingBufferColorSpaceInString}`);
+        gl.drawingBufferColorSpace = drawingBufferColorSpace;
+      }
+      as_expected = expectedDrawingBufferColorSpace;
+      shouldBe('gl.drawingBufferColorSpace', 'as_expected');
+    }
+
+    gl.clearColor(color[0], color[1], color[2], 1.0);
+    gl.clear(gl.COLOR_BUFFER_BIT);
+    gl.flush();
+
+    shouldBe('gl.getError()', 'gl.NO_ERROR');
+
+    checkPixels(gl, w, h, color, 1);
+  }
+
+  canvas.remove();
```

Comment:
I think it's customary to leave the element to the canvas, so when the test fails it's easier to debug?

Reply:
I found that there's a limit of 15 webgl canvas, so I was trying not to keep them all around. But I think the test would still work even if old canvases become inactive, the text results should stay the same.